### PR TITLE
Bump Net461 to Net462

### LINKED ISSUE #4315: [Engineering Task] Bump net 461 to net 462
### Type of task?

Supportability

### Description

Bump net 461 version to net 462 in MSAL.

### Solution

_No response_

### PR DESCRIPTION
**Pull request by neha-bhargava**
Fixes #4315 

**Changes proposed in this request**
Bump Net461 to 462

**Testing**

**Performance impact**

**Documentation**
- [ ] All relevant documentation is updated.

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="MSBuild.Sdk.Extras">
 
   <PropertyGroup Label="Targets required for unit tests to run">
-    <TargetFrameworkNetDesktop461>net461</TargetFrameworkNetDesktop461>
+    <TargetFrameworkNetDesktop461>net462</TargetFrameworkNetDesktop461>
```
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj:4]: Update the element name for consistency - `TargetFrameworkNetDesktop462`

**Comment by gladjohn** [comment]
just the client net462 bump? or should we also bump desktop and broker as well?

File: src/client/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj (hunk 1 above)
Comment (gladjohn) [reply src/client/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj:4]: that would also need an update to `platform_and_feature_flags.props`

**Comment by bgavrilMS** [comment]
> just the client net462 bump? or should we also bump desktop and broker as well?

Yes, I think we should. It will be easier to maintain.

**Review by pmaytak — APPROVED** [review]

**Review by pmaytak — CHANGES_REQUESTED** [review]
There may be some issues. Create a net461 app and add Msal.Desktop package. It gives errors.
image

**Review by gladjohn — CHANGES_REQUESTED** [review]
let's test both broker and desktop dev apps with the change

**Comment by neha-bhargava** [comment]
Updated and tested both using dev app.

**Review by gladjohn — APPROVED** [review]

File: src/client/Microsoft.Identity.Client.Desktop/Microsoft.Identity.Client.Desktop.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">
   <PropertyGroup>
 
-    <TargetFrameworkNetDesktop>net461</TargetFrameworkNetDesktop>
+    <TargetFrameworkNetDesktop>net462</TargetFrameworkNetDesktop>
```
[the remark this answers is not in the document]
Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client.Desktop/Microsoft.Identity.Client.Desktop.csproj:4]: We should not cater for net461 apps. Apps must be on supported version of .NET, which is net462

**Review by pmaytak — APPROVED** [review]

**Pull request merged**